workers/go/kitchensink/kitchen_sink.go

```diff
@@ -74,141 +76,90 @@ func handleActionSet(
 
 func handleAction(
 	ctx workflow.Context,
-	params *kitchensink.WorkflowParams,
 	action *kitchensink.Action,
 ) (shouldReturn bool, returnValue interface{}, err error) {
-	info := workflow.GetInfo(ctx)
-	switch {
-	case action.Result != nil:
-		if action.Result.RunID {
-			return true, info.WorkflowExecution.RunID, nil
-		}
-		return true, action.Result.Value, nil
-
-	case action.Error != nil:
-		if action.Error.Attempt {
-			return true, nil, fmt.Errorf("attempt %v", info.Attempt)
-		}
-		var details []interface{}
-		if action.Error.Details != nil {
-			details = append(details, action.Error.Details)
-		}
-		return true, nil, temporal.NewApplicationError(action.Error.Message, "", details...)
-
-	case action.ContinueAsNew != nil:
-		if action.ContinueAsNew.WhileAboveZero > 0 {
-			action.ContinueAsNew.WhileAboveZero--
-			return true, nil, workflow.NewContinueAsNewError(ctx, KitchenSinkWorkflow, params)
-		}
-
-	case action.Sleep != nil:
-		if err := workflow.Sleep(ctx, time.Duration(action.Sleep.Millis)*time.Millisecond); err != nil {
+	if rr := action.GetReturnResult(); rr != nil {
+		return true, rr.ReturnThis, nil
+	} else if re := action.GetReturnError(); re != nil {
+		return true, nil, temporal.NewApplicationError(re.Failure.Message, "")
+	} else if can := action.GetContinueAsNew(); can != nil {
+		return true, nil, workflow.NewContinueAsNewError(ctx, KitchenSinkWorkflow, can.Arguments)
+	} else if timer := action.GetTimer(); timer != nil {
+		if err := workflow.Sleep(ctx, time.Duration(timer.Milliseconds)*time.Millisecond); err != nil {
```

Comment:
Timer action has an `AwaitableChoice` that is ignored

Reply:
Oh do we never handle  `AwaitableChoice` in the Go workflow?

Reply:
It's not handled in the python one either yet. This is one of the things I'll add in another PR.

Reply:
Works for me, to make my life easier can you mark the stuff you plan to do in follow up PRs with `TODO` or something? It is hard to tell if something is planned to be done but just hasn't, was forgotten, or I misunderstood a feature.

Reply:
That's going to be quite a lot of stuff - essentially everything in the DSL that isn't currently emitted anywhere by the generator.

If something's in the generator, the workflows need to handle it since it could be generated. I think the generator is reasonably easy to read. Does that work?

Reply:
I understand, yeah I can work with that